clib/src/main/resources/scala-native/stdio.c

```diff
@@ -0,0 +1,36 @@
+#include <stdio.h>
```

Comment:
I think this is a concrete case which captures some architectural disquiet I have been 
trying to put into sensible words.

From build.sbt it appears that the intent is that javalib & clib be independent and rely upon nativelib.  From the description of this PR, it appears that the intent is that nativelib be minimal/skinny/lightweight.  So far so good.

Now we come upon stdio.  It is defined in both posix and c.  To the best of my searching in a clone of the PR branch, there is a stdio.scala in clib (corresponding, I believe, to this C file) but _none_ in posix lib, where a posix developer would expect to find it.

There are at least four possibilities, none good, and I think it takes an architectural
decision in a README in each of posixlib and clib:

0) Let posixlib have holes as an explicit decision.

1) code duplication.  posixlib & clib maintain their own bugs and quirks. (Is my
    disregard for this solution showing? Oops!)

2) Posixlib calls into clib.  This violates one of the inferred goals of this PR.

3) Posixlib and clib call into a common lower layer. This gets up back to nativelib
    or such, which violates another intent.

Reply:
I think what you have above should definitely be captured in an issue and when determined in the developer docs. This is a tricky issue. I have thought about it some but don't know the answer or even the balance of the options.

Reply:
Even under its own admission POSIX extends the C standard - click on the [CX] to get some more info.
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stdio.h.html